tests/drivers/flash/src/main.c

```diff
@@ -91,9 +91,10 @@ static void *flash_driver_setup(void)
 	}
 
 	if (!is_buf_clear) {
-		/* erase page */
+		/* Erase a nb of pages aligned to the EXPECTED_SIZE */
 		rc = flash_erase(flash_dev, page_info.start_offset,
-				 page_info.size);
+				(page_info.size * (EXPECTED_SIZE / page_info.size + 1)));
```

Comment:
I think that correct math is `(page_info.size * ((EXPECTED_SIZE + page_info.size - 1) / page_info.size)`